Personal user space code and keymaps by ZyBeR

ZyberSE opened the pull request:
My personal user space code and keymaps for some of the boards I own.
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
This contains various bits of code in my user space section and several keymaps for different boards. I also fixed a compile warning in knife66/knife66_iso for the new deprecated parameter RGBLIGHT_EFFECT_KNIGHT. No changes outside of that is included in this PR.

<!--- Describe your changes in detail here. -->
My keymaps are made to be used on macOS with the "swansi" keymap and ansi layout. Some of the features are TapDaces to write Swedish accent chars, double tap 4 to take a screenshoot and keep it in the clipboard and some boards have a function to paste "code blocks" to Discord, Slack etc by just tapping a single button which triggers a macro which outputs the required back-ticks and pasts the clipboard content.
## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

fauxpark commented on keyboards/playkbtw/ca66/keymaps/zyber/rules.mk:
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no
+MOUSEKEY_ENABLE = no	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no		# Console for debug
+COMMAND_ENABLE = no		# Commands for debug and configuration
+SLEEP_LED_ENABLE = no	# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
```suggestion
MOUSEKEY_ENABLE = no
```

fauxpark commented on keyboards/wilba_tech/zeal65/rules.mk:
```diff
@@ -20,7 +20,7 @@ COMMAND_ENABLE = no        # Commands for debug and configuration
 BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
 AUDIO_ENABLE = no           # Audio output
-RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
```
```suggestion
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight. 
```

ZyberSE replied:
Why leave the trailing white space?

fauxpark replied:
That can be fixed in another PR. Try to keep keyboard changes separate from keymap/userspace changes.

ZyberSE replied:
OK

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+
+		case CTRL_C_UP:
+			if (record->event.pressed) {
+				register_mods(MOD_BIT(KC_LCTRL));
```
```suggestion
				register_mods(MOD_BIT(KC_LEFT_CTRL));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+			if (record->event.pressed) {
+				register_mods(MOD_BIT(KC_LCTRL));
+			} else {
+				unregister_mods(MOD_BIT(KC_LCTRL));
```
```suggestion
				unregister_mods(MOD_BIT(KC_LEFT_CTRL));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+/* Screenshoot */
+void dance_SSHT_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LGUI);
+		register_code(KC_LCTRL);
+		register_code(KC_LSHIFT);
+		register_code(KC_4);
+		unregister_code(KC_LGUI);
+		unregister_code(KC_LCTRL);
+		unregister_code(KC_LSHIFT);
+		unregister_code(KC_4);
```
```suggestion
                tap_code16(C(S(G(KC_4))));
```

ZyberSE replied:
OMG, thank you I didn't know I could do this :)

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+		unregister_code(KC_LSHIFT);
+		unregister_code(KC_4);
+	} else {
+		register_code(KC_4);
+		unregister_code(KC_4);
```
```suggestion
		tap_code(KC_4);
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+}
+void dance_SSHT_reset(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		unregister_code(KC_LGUI);
+		unregister_code(KC_LCTRL);
+		unregister_code(KC_LSHIFT);
+		unregister_code(KC_4);
```
```suggestion
		unregister_code16(C(S(G(KC_4))));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+/* Å */
+void dance_LBRC_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_LBRC);
+		unregister_code(KC_LALT);
+		unregister_code(KC_LBRC);
```
```suggestion
                tap_code16(A(KC_LBRC));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+		unregister_code(KC_LALT);
+		unregister_code(KC_LBRC);
+	} else {
+		register_code(KC_LBRC);
+		unregister_code(KC_LBRC);
```
```suggestion
		tap_code(KC_LBRC);
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+}
+void dance_LBRC_reset(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		unregister_code(KC_LALT);
+		unregister_code(KC_LBRC);
```
```suggestion
		unregister_code16(A(KC_LBRC));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+/* Ö */
+void dance_SCLN_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_SCLN);
+		unregister_code(KC_LALT);
+		unregister_code(KC_SCLN);
```
```suggestion
		tap_code16(A(KC_SCLN));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+		unregister_code(KC_LALT);
+		unregister_code(KC_SCLN);
+	} else {
+		register_code(KC_SCLN);
+		unregister_code(KC_SCLN);
```
```suggestion
		tap_code(KC_SCLN);
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+/* Ä */
+void dance_QUOT_finished(qk_tap_dance_state_t *state, void *user_data) {
+	if (state->count == 2) {
+		register_code(KC_LALT);
+		register_code(KC_QUOT);
+		unregister_code(KC_LALT);
+		unregister_code(KC_QUOT);
```
```suggestion
		tap_code16(A(KC_QUOT));
```

fauxpark commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,160 @@
+		unregister_code(KC_LALT);
+		unregister_code(KC_QUOT);
+	} else {
+		register_code(KC_QUOT);
+		unregister_code(KC_QUOT);
```
```suggestion
		tap_code(KC_QUOT);
```

ZyberSE commented:
The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`.
What can I do to resolve this?

tzarc commented:
> The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`. What can I do to resolve this?

This is now fixed.

The other issue along those lines is that you've made modifications to the `config.h` files for both the knife66 keyboards -- you'll need to revert those changes.

ZyberSE commented:
> > The CI build complains about `misterknife/knife66_iso:zyber` issues is I do not have a keymap in that folder. Under misterknife there are two folders, one `knife66` and one `knife66_iso`. My keymap is in `knife66`. What can I do to resolve this?
> 
> This is now fixed.
> 
> The other issue along those lines is that you've made modifications to the `config.h` files for both the knife66 keyboards -- you'll need to revert those changes.

Not a problem - I'll revert those, but the original code generates compiler warnings due to the deprecated option I removed.

drashna commented on keyboards/ai03/lunar/keymaps/zyber/config.h:
```diff
@@ -0,0 +1,5 @@
+/**
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Also, you want/need an include guard here (`#pragma once`)

drashna commented on keyboards/alps64/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,31 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```


Also, if you have `#include QMK_KEYBOARD_H` in your zyber.h file, you could just include that here instead.

drashna commented on keyboards/converter/adb_usb/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on keyboards/fc660c/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,45 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on keyboards/playkbtw/ca66/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,19 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/config.h:
```diff
@@ -0,0 +1,5 @@
+// USB Device descriptor parameter
+#undef PRODUCT
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

And an `#pragma once`

drashna commented on keyboards/wilba_tech/rama_works_m60_a/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,41 @@
+// M60-A layout
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on keyboards/wilba_tech/zeal65/keymaps/zyber/config.h:
```diff
@@ -0,0 +1,28 @@
+#pragma once
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on keyboards/wilba_tech/zeal65/keymaps/zyber/keymap.c:
```diff
@@ -0,0 +1,38 @@
+// ZyBeRs layout for Zeal65
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on users/zyber/config.h:
```diff
@@ -0,0 +1,8 @@
+#define GRAVE_ESC_GUI_OVERRIDE
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

and `#pragma once`

drashna commented on users/zyber/zyber.c:
```diff
@@ -0,0 +1,134 @@
+#include "zyber.h"
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

drashna commented on users/zyber/zyber.h:
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```
